### ISSUE TITLE
Work around solution for getting actual temporary key values from the entity's tracking information failed for non-foreign key column.

### ISSUE THREAD

**Issue by Nivitha-Gopalakrishnan** [dotnet/efcore#19326]
As in our application, we have a situation where we used to insert data-parallel in two table and finally save changes of the context at the end and faced problem due to this breaking changes in EF Core 3.0 as mention here (https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-3.0/breaking-changes#temporary-key-values-are-no-longer-set-onto-entity-instances). However we resolve this problem by applying for the suggested workaround solution to get the temporary key values from the below query, which returns the "Id", primary column temporary value for "books" table

var  tempIdValue = context.Entry(books).Property(e => e.Id).CurrentValue

This solution works fine when two tables have foreign key references relation whereas it failed and inserted the actual temporary value (i.e. long negative numbers) in the column if they don’t have foreign key references.

For Example,

Let us consider Books and Students are two master tables for library-related applications and when track books borrowed from the library by a student in the "StudentBookMapper" table.

**Database Schema:**

1. Table Name - Students

**Columns:**
Id
Name
Place
Country

**Note:** Id is the Primary Key

2. Table Name - Books

**Columns:**
Id
Name
Category
Author

3. Table Name - StudentBookMapper

**Columns:**

Id
UserId
BookId
IsActive

Here "BookId" column is a foreign key reference of the "Books" table with the "Id" column similarly "UserId" column is a foreign key reference of the "Students" table with "Id". When we try to insert a new entry in "Students" and along the "StudentBookMapper" table using the above temporary value query everything works fine. But now we got a situation where we need to keep the PG Student book borrowed details in the same table and hence removed the foreign key reference. 

Henceforth, the column "UserId" corresponds to both the "Students" table and "PGStudents" table with a differentiate column "IsPGStudent" in "StudentBookMapper".

Modified Table Scheme for Table - StudentBookMapper

**Columns:**

Id
UserId
BookId
IsPGStudent
IsActive

In this scenario when we try the above temporary value query to insert the details in "Students" and "StudentBookMapper" table (i.e. "UserId" column is not foreign key referenced with "Students" table) we get a result where the actual temporary value (i.e. long negative value) inserted whereas the expected value to be inserted is "Primary column" value of "Students" table. 

**Note:** We do save changes of the context once the data added to both the tables not once added to.

Kindly let me know is this an issue or anything wrong in our approach.
 
### Further technical details

EF Core version: 3.1
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1
IDE: Visual Studio 2019 16.4.1

**Comment by ajcvickers** [comment]
@Nivitha-Gopalakrishnan While I appreciate the detailed description it is still hard to follow exactly what is going on. It would be much more useful to get a small, runnable project or code listing that demonstrates the behavior you are seeing so that we can investigate.

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by Nivitha-Gopalakrishnan** [comment]
I have created the sample for this. Kindly check it and let me know.
